### DISCUSSION TITLE
How to get success/fail response?

### DISCUSSION THREAD

**Discussion by harshpawar** [jofftiquez/vue-stripe-checkout#190] [Q&A]
I am using the checkout method.
I could not found any method or documentation regarding to the response getting back from stripe.
like success response or fail response of the payment.
Please suggest a good way or method regarding to get response in VUE pages.

**Comment by jofftiquez** [comment]
Hi, you are probably looking for the `successUrl`, and the `cancelUrl`. Both are documented [here](https://vuestripe.com/stripe-checkout/getting-started), under the props section.

To give you a tip, here's what I did on my app.

I set the current page's URL as `successUrl` and the `cancelUrl` but with different query params to determine if the payment was successful or not upon returning from checkout. 

```javascript
this.successUrl = `${window.location.origin}${window.location.pathname}?state=success`;
this.cancelUrl = `${window.location.origin}${window.location.pathname}?state=cancel`;
```
And then I catch the query params upon the loading of the page to see whether the value of the `$route.query.state` is either `success` or `cancel`.

--------------------------

You can see the exact behavior in the documentation of Vue Stripe Checkout [here](https://vuestripe.com/stripe-checkout/one-time-payment).

You can also check the code [here](https://github.com/vue-stripe/vuestripe.com/blob/328e2921f335c5896cb5b4d6e1f6391bb2391549/pages/stripe-checkout/one-time-payment.vue#L84).

**Reply by harshpawar** [reply]
That is i know.
but it only returns success or fail or redirect on specific page.
what about the reference id, payment_id or as i am using the subscription then i also need to store the subscription id related fields so whenever user need to cancel the subscription i can use it.
we did not get any response JSON or anything which contain payment related details.
like payment_id, reference_id, date and time and amount, and all.

**Reply by jofftiquez** [reply]
You can check Stripe webhooks for that.
